### PR TITLE
Inline inputs for checkbox and radio buttons

### PR DESCRIPTION
**Pull request by delisma**
CSS changes and coded documentation for inline checkboxes and radio buttons inputs in `<div>` elements and `<li>` elements

### CODE REVIEW

File: src/plugins/formvalid/_base.scss [hunk 1]
```diff
@@ -46,5 +46,31 @@
...
+		&.h5 {
+			float: none;
+			margin-bottom: 0;
+			.error {
+				margin-top: 1em;
+			}
+		}
 	}
 }
+ul.form-group {
```
Comment (duboisp) [inline src/plugins/formvalid/_base.scss]: This style do not belong to the formvalid plugin, instead it should be more global, like the bootstrap override

File: Gruntfile.coffee [hunk 2]
```diff
@@ -1009,6 +1009,7 @@ module.exports = (grunt) ->
 						"W010" # Using `.pull-left` or `.pull-right` as part of the media object component is deprecated as of Bootstrap v3.3.0. Use `.media-left` or `.media-right` instead.
 						"E013" # Only columns (`.col-*-*`) may be children of `.row`s
 						"E014" # Columns (`.col-*-*`) can only be children of `.row`s or `.form-group`s
+						"E017" # Support for checkboxes with explicit labels instead of implicit labels
```
Comment (duboisp) [inline Gruntfile.coffee]: We will move this exception for the formvalid page only.

File: package.json [hunk 3]
```diff
@@ -34,6 +34,7 @@
   "license": "MIT",
   "dependencies": {
     "bootstrap-sass": "3.4.1",
+    "bower": "^1.8.8",
```
Comment (GormFrank) [inline package.json]: Please do not include package.json and package-lock.json to your PR.

File: src/plugins/formvalid/_base.scss (hunk 1 above)
Comment (GormFrank) [inline src/plugins/formvalid/_base.scss]: Put those styles inside another file, do not encapsulate them inside formvalid

File: Gruntfile.coffee (hunk 2 above)
Comment (GormFrank) [inline Gruntfile.coffee]: We'll have to look into a new rule for this use case. (checkbox AND radio included)

**Review by GormFrank — CHANGES_REQUESTED** [review]

**Review by duboisp — CHANGES_REQUESTED** [review]
The update to package.json and package-lock.json would need to be removed from this PR.

File: Gruntfile.coffee (hunk 2 above)
Comment (EricDunsworth) [reply Gruntfile.coffee]: In case it helps: What @duboisp is suggesting would be very similar to what I recently did in wet-boew/GCWeb#1605

**Review by duboisp — COMMENTED** [review]
Todo:
* Test accessibility - Check if the design pattern that use unordered list of "radio" or "checkbox" is acceptable
* Test accessibility - Check if the design pattern that use ```<ul>``` inside a ```fieldset``` is too much
* Test accessibility - Check if we only use ```div``` for a list of ```input[type=radio]``` and  ```input[type=radio]```
* Test accessibility - Is radio and checkbox must be always in a fieldset?
* Test accessibility - When radio and checkbox are not in a fieldset, can they be put in a list

Bootlint don't seems to support explicit labels. Explicit labels is needed to enable a improved visual for bigger and flat checkbox/radio that would improve UX.

**Comment by duboisp** [comment]
@bsouster as discussed, you will find the pattern in the last example in this demo
https://universallabs.org/wet/labs/pr8744/unmin/demos/formvalid/formvalid-en.html

Also, see my [previous comments](https://github.com/wet-boew/wet-boew/pull/8744#pullrequestreview-305986836) regarding what we need to test and validate

**Comment by EricDunsworth** [comment]
FWIW this is how Bootstrap 3.4's documentation recommends doing layouts for horizontal checkboxes/radio buttons:
https://getbootstrap.com/docs/3.4/css/#inline-checkboxes-and-radios

File: src/base/bootstrap-overrides/_base.scss [hunk 4]
```diff
@@ -589,6 +589,24 @@ input[type=button], input[type=reset], input[type=submit] {
...
 
+/*
+ * Explicit labels for checkbox and radio button inputs
+ *
+ * Allow the use of horizontal explicit labels instead of the default implicit
+ * labelling pattern for checkbox and radio button inputs provided by Bootstrap.
+ */
+.form-inline {
+	.label-inline {
+		font-weight: 400;
```
Comment (GormFrank) [inline src/base/bootstrap-overrides/_base.scss]: There might be a variable available for this one.

File: src/base/bootstrap-overrides/_base.scss [hunk 5]
```diff
@@ -589,6 +589,24 @@ input[type=button], input[type=reset], input[type=submit] {
...
+		margin-bottom: 0;
+		padding-left: 1px;
+		vertical-align: middle;
+	}
+	.form-group + .form-group {
+		padding-left: 10px;
```
Comment (GormFrank) [inline src/base/bootstrap-overrides/_base.scss]: The mrgn-lft-X class should be used instead.

File: src/base/bootstrap-overrides/_base.scss [hunk 6]
```diff
@@ -648,6 +666,9 @@ fieldset {
 			font-weight: bold;
 			margin-bottom: 5px;
 		}
+		input + label {
+			padding: 0 1ch;
```
Comment (GormFrank) [inline src/base/bootstrap-overrides/_base.scss]: Update this one

**Review by GormFrank — CHANGES_REQUESTED** [review]
Also don't forget to squash your commits.

File: src/base/bootstrap-overrides/_base.scss (hunk 5 above)
Comment (delisma) [reply src/base/bootstrap-overrides/_base.scss]: The class mrgn-lft-x adds margin not padding.

File: src/base/bootstrap-overrides/_base.scss (hunk 4 above)
Comment (delisma) [reply src/base/bootstrap-overrides/_base.scss]: I've checked all the scss files to find a variable that set a font-weight to normal but nothing seems to fit the needs.

File: src/base/bootstrap-overrides/_base.scss (hunk 6 above)
Comment (delisma) [reply src/base/bootstrap-overrides/_base.scss]: Done

**Comment by GormFrank** [comment]
Pre-approved during technical review conditionally to testing and @duboisp accepting changes.

**Review by GormFrank — APPROVED** [review]

File: src/base/bootstrap-overrides/_base.scss [hunk 7]
```diff
@@ -589,6 +589,35 @@ input[type=button], input[type=reset], input[type=submit] {
...
+		position: relative;
+		vertical-align: middle;
+		label {
+			font-weight: 400;
+			margin-bottom: 0px;
+			padding-left: 2px;
+		}
+	}
+}
+@media (min-width: $screen-sm-min) {
```
Comment (duboisp) [inline src/base/bootstrap-overrides/_base.scss]: Move this in the view or create one for bootstrap and add it the main SCSS include file

**Review by duboisp — CHANGES_REQUESTED** [review]

**Review by duboisp — APPROVED** [review]

**Comment by duboisp** [comment]
Pre-approved waiting for Travis-CI biuld to pass.

**Pull request merged**